src/industry_cmd.cpp

```diff
@@ -2891,7 +2892,7 @@ void IndustryMonthlyLoop()
 	for (Industry *i : Industry::Iterate()) {
 		UpdateIndustryStatistics(i);
 		if (i->prod_level == PRODLEVEL_CLOSURE) {
-			delete i;
+			if (!(i->ctlflags & INDCTL_NO_CLOSURE)) delete i;
```

Comment:
This is already checked before PRODLEVEL_CLOSURE is assigned, isn't it?

Reply:
What should happen if the GS sets the flag after the industry has announced closure but before it gets deleted?

Reply:
IMO, when a closure is started, it should be completed.